apps/accounts/models.py

```diff
@@ -97,22 +98,22 @@ def save(self, *args, **kwargs):
         if not self.valence:
             self.valence = self.emotion.valence
 
+        self.full_clean()
+
         super().save(*args, **kwargs)
 
-    def update_emotion_boundaries(self, valence, energy, reset=False):
+    def update_attributes(self):
         """
-        Given the valence and energy of a song, recompute boundaries for the given emotion box
-        :param valence: (float) Representation of song mood
-        :param energy: (float) Representation of song intensity
-        :param change: (bool) Flag to denote resetting the boundaries for a record. Used in the case a user "unvotes"
-        a song to reset the boundaries for that emotion
+        Update the attributes for this user/emotion mapping to the average of the attributes of the songs the user has
+        upvoted as making them feel this emotion. If the user doesn't have any upvotes for this emotion, the attributes
+        will be set to `None` and reset to the emotion defaults in the save() call
         """
-        if reset:
-            self.valence = 2 * self.valence - valence
-            self.energy = 2 * self.energy - energy
-        else:
-            self.valence = (self.valence + valence) / 2
-            self.energy = (self.energy + energy) / 2
+        votes = self.user.usersongvote_set.filter(emotion=self.emotion, vote=True)
+        valences = votes.values_list('song__valence', flat=True)
+        energies = votes.values_list('song__energy', flat=True)
+
+        self.valence = average(valences)
```

Comment:
Instead of calculating the new attributes based on average, do we want to instead set it to the median value of the attribute values? This could help avoid outliers over-influencing the value (think average income calculations and the top 1% throwing off the data for the population)

Reply:
After talking to another data scientist, using average to find the middle point is not a bad choice. This is described as the "average point", the middle datapoint of all records in a set. Using the mean to find this value is not out of the question, and serves our purposes well.